pkg/providers/vsphere/validator.go

```diff
@@ -580,29 +614,47 @@ func (v *Validator) validateCPUserPrivs(ctx context.Context, spec *Spec, vuc *co
 		datacenter,
 	)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	return v.validatePrivs(ctx, privObjs, vsc)
 }
 
-func (v *Validator) validatePrivs(ctx context.Context, privObjs []PrivAssociation, vsc govmomi.VSphereClient) error {
-	var missingPrivs []string
+func (v *Validator) validatePrivs(ctx context.Context, privObjs []PrivAssociation, vsc govmomi.VSphereClient) (bool, error) {
+	var privs []string
 	var err error
+	missingPrivs := []missingPriv{}
+	passed := false
 
 	for _, obj := range privObjs {
 		path := obj.path
 		privsContent := obj.privsContent
 		t := obj.objectType
-		missingPrivs, err = v.getMissingPrivs(ctx, vsc, path, t, privsContent, vsc.Username())
+		username := vsc.Username()
+		privs, err = v.getMissingPrivs(ctx, vsc, path, t, privsContent, username)
 		if err != nil {
-			return err
-		} else if len(missingPrivs) > 0 {
-			return fmt.Errorf("User %s missing privileges on %s: %v", vsc.Username(), path, missingPrivs)
+			return passed, err
+		} else if len(privs) > 0 {
+			mp := missingPriv{
+				Username:    username,
+				ObjectType:  t,
+				Path:        path,
+				Permissions: privs,
+			}
+			missingPrivs = append(missingPrivs, mp)
+			content, _ := yaml.Marshal(mp)
```

Comment:
We may still want to catch the error then but maybe say "Warning: failed to list missing privs: %v"?